venue/tasks.py

```diff
@@ -54,110 +70,128 @@ def get_expected_links(code):
+            posts = scraper.scrape_posts(
+                forum_profile.forum_user_id,
+                start=posts_scrape_start.replace(tzinfo=None)
             )
-            if not post_check.exists():
-                forum_post = ForumPost(
-                    user_profile=forum_profile.user_profile,
+            # Save each new post
+            message_ids = []
+            for post in posts:
+                message_ids.append(post['message_id'])
+                post_check = ForumPost.objects.filter(
                     forum_profile=forum_profile,
-                    forum_rank=forum_profile.forum_rank,
                     topic_id=post['topic_id'],
                     message_id=post['message_id'],
-                    unique_content_length=post['content_length'],
-                    timestamp=post['timestamp']
                 )
-                forum_post.save()
-        # Check for post deletion
-        deleted_posts = ForumPost.objects.filter(
-            forum_profile=forum_profile,
-            matured=False,
-            timestamp__gte=posts_scrape_start
-        ).exclude(
-            message_id__in=message_ids
-        )
-        for post in deleted_posts:
-            post.credited = False
-            post.monitoring = False
-            post.save()
-        # Update tracked posts
-        for post in tracked_posts:
-            forum_post = ForumPost.objects.get(
-                message_id=post,
-                forum_profile=forum_profile
+                if not post_check.exists():
+                    forum_post = ForumPost(
+                        user_profile=forum_profile.user_profile,
+                        forum_profile=forum_profile,
+                        forum_rank=forum_profile.forum_rank,
+                        topic_id=post['topic_id'],
+                        message_id=post['message_id'],
+                        unique_content_length=post['content_length'],
+                        timestamp=post['timestamp']
+                    )
+                    forum_post.save()
+            # Check for post deletion
+            deleted_posts = ForumPost.objects.filter(
+                forum_profile=forum_profile,
+                matured=False,
+                timestamp__gte=posts_scrape_start
+            ).exclude(
+                message_id__in=message_ids
             )
-            forum_post.save()
-        # Update the forum_profile's last scrape timestamp
-        forum_profile.last_scrape = timezone.now()
-        forum_profile.save()
+            for post in deleted_posts:
+                post.credited = False
+                post.monitoring = False
+                post.save()
+            # Update tracked posts
+            for post in tracked_posts:
+                forum_post = ForumPost.objects.get(
+                    message_id=post,
+                    forum_profile=forum_profile
+                )
+                forum_post.save()
```

Comment:
What is the point of the code block 181- 186?

Reply:
It's meant to trigger the `ForumPost`'s post save logic for each post. This one: https://github.com/Volentix/venue-server/blob/4b4cd81accf81bed3aed4af0b614ec3bb97cec8c/venue/models.py#L423-L471